lock-api/src/main/java/com/palantir/lock/client/timestampleases/ExactTimestampSupplier.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client.timestampleases;
+
+import com.palantir.atlasdb.timelock.api.ConjureTimestampRange;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.util.function.LongSupplier;
+
+final class ExactTimestampSupplier implements LongSupplier {
```

Comment:
Should we just call this CountingLongSupplier?

Reply:
Or LimitedCountingSupplier?

Reply:
Then, compose that with ConjureTImestampRangeTImestampSupplier? Or something. Basically it feels like you have multiple responsibilities in this class

Reply:
I think Sam can reuse the LimitedLongSupplier as well in this way